### ISSUE TITLE
netapp.ontap.na_ontap_interface is not idempotent

### ISSUE THREAD

**Issue by JamesPGriffith** [ansible-collections/netapp.ontap#91]
### Summary

netapp.ontap.na_ontap_interface re-runs are failing because the LIF already exists with the IP. This is not idempotent.

### Component Name

netapp.ontap.na_ontap_interface

### Ansible Version

```console
user@jumphost:~/ontap-setup-automation$ ansible --version
ansible [core 2.13.2]
  config file = /home/user/ontap-setup-automation/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/.local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/user/.local/bin/ansible
  python version = 3.8.10 (default, Jun 22 2022, 20:18:18) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True
user@jumphost:~/ontap-setup-automation$
```

### ONTAP Collection Version

```console
user@jumphost:~/ontap-setup-automation$ ansible-galaxy collection list

# /home/user/.local/lib/python3.8/site-packages/ansible_collections
Collection                    Version
----------------------------- -------
amazon.aws                    3.3.1  
ansible.netcommon             3.0.1  
ansible.posix                 1.4.0  
ansible.utils                 2.6.1  
ansible.windows               1.10.0 
arista.eos                    5.0.1  
awx.awx                       21.2.0 
azure.azcollection            1.13.0 
check_point.mgmt              2.3.0  
chocolatey.chocolatey         1.3.0  
cisco.aci                     2.2.0  
cisco.asa                     3.1.0  
cisco.dnac                    6.5.0  
cisco.intersight              1.0.19 
cisco.ios                     3.2.0  
cisco.iosxr                   3.2.0  
cisco.ise                     2.5.0  
cisco.meraki                  2.10.0 
cisco.mso                     2.0.0  
cisco.nso                     1.0.3  
cisco.nxos                    3.1.0  
cisco.ucs                     1.8.0  
cloud.common                  2.1.2  
cloudscale_ch.cloud           2.2.2  
community.aws                 3.4.0  
community.azure               1.1.0  
community.ciscosmb            1.0.5  
community.crypto              2.4.0  
community.digitalocean        1.21.0 
community.dns                 2.2.1  
community.docker              2.7.0  
community.fortios             1.0.0  
community.general             5.3.0  
community.google              1.0.0  
community.grafana             1.5.0  
community.hashi_vault         3.0.0  
community.hrobot              1.4.0  
community.libvirt             1.1.0  
community.mongodb             1.4.1  
community.mysql               3.3.0  
community.network             4.0.1  
community.okd                 2.2.0  
community.postgresql          2.1.5  
community.proxysql            1.4.0  
community.rabbitmq            1.2.1  
community.routeros            2.1.0  
community.sap                 1.0.0  
community.sap_libs            1.1.0  
community.skydive             1.0.0  
community.sops                1.2.3  
community.vmware              2.7.0  
community.windows             1.10.0 
community.zabbix              1.7.0  
containers.podman             1.9.4  
cyberark.conjur               1.1.0  
cyberark.pas                  1.0.14 
dellemc.enterprise_sonic      1.1.1  
dellemc.openmanage            5.5.0  
dellemc.os10                  1.1.1  
dellemc.os6                   1.0.7  
dellemc.os9                   1.0.4  
f5networks.f5_modules         1.18.0 
fortinet.fortimanager         2.1.5  
fortinet.fortios              2.1.6  
frr.frr                       2.0.0  
gluster.gluster               1.0.2  
google.cloud                  1.0.2  
hetzner.hcloud                1.8.1  
hpe.nimble                    1.1.4  
ibm.qradar                    2.0.0  
infinidat.infinibox           1.3.3  
infoblox.nios_modules         1.3.0  
inspur.sm                     2.0.0  
junipernetworks.junos         3.1.0  
kubernetes.core               2.3.2  
mellanox.onyx                 1.0.0  
netapp.aws                    21.7.0 
netapp.azure                  21.10.0
netapp.cloudmanager           21.18.0
netapp.elementsw              21.7.0 
netapp.ontap                  21.20.0
netapp.storagegrid            21.10.0
netapp.um_info                21.8.0 
netapp_eseries.santricity     1.3.0  
netbox.netbox                 3.7.1  
ngine_io.cloudstack           2.2.4  
ngine_io.exoscale             1.0.0  
ngine_io.vultr                1.1.2  
openstack.cloud               1.8.0  
openvswitch.openvswitch       2.1.0  
ovirt.ovirt                   2.1.0  
purestorage.flasharray        1.13.0 
purestorage.flashblade        1.9.0  
purestorage.fusion            1.0.2  
sensu.sensu_go                1.13.1 
servicenow.servicenow         1.0.6  
splunk.es                     2.0.0  
t_systems_mms.icinga_director 1.30.0 
theforeman.foreman            3.4.0  
vmware.vmware_rest            2.2.0  
vyos.vyos                     3.0.1  
wti.remote                    1.0.4  

# /home/user/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 5.4.0  
netapp.ontap      21.22.0

# /usr/share/ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 5.4.0  
user@jumphost:~/ontap-setup-automation$
```

### ONTAP Version

```console
cluster1::> version
NetApp Release 9.8P10: Fri Feb 04 19:51:21 UTC 2022

cluster1::>
```

### Playbook

```console
- name: Cluster Mgmt LIF
  na_ontap_interface:
    state: present
    interface_name: "{{ cluster_mgmt.interface_name }}"
    home_port: "{{ cluster_mgmt.home_port }}"
    home_node: "{{ cluster_mgmt.home_node }}"
    ipspace: "{{ cluster_mgmt.ipspace }}"
    address: "{{ cluster_mgmt.address }}"
    netmask: "{{ subnet }}"
    hostname: "{{ node1_ip }}"
    username: "{{ netapp_username }}"
    password: "{{ netapp_password }}"
    https: true
    validate_certs: "{{ validate_certs }}"
```

### Steps to Reproduce

./globalvar.yml
```yaml
# globalvars.yml
---
# Lab variables
domain_name: demo.netapp.com
admin_user_name: administrator
admin_password: Netapp1!
netapp_username: admin
netapp_password: Netapp1!
dns:
    domain: demo.netapp.com
    nameserver: 192.168.0.253
ntp:
  server_name: 192.168.0.1
  version: auto
subnet: 255.255.255.0
gateway: 192.168.0.1
validate_certs: False
```
./cluster1.yml
```yaml
---
cluster: cluster1
netapp_hostname: 192.168.0.101
cluster_mgmt:
  interface_name: cluster1_mgmt
  # address: '{{ "lookup('community.general.dig', '{{cluster}}.{{domain_name}}')"}}'
  address: 192.168.0.111
  home_node: "{{cluster}}-01"
  home_port: e0M
  ipspace: Default
node1_ip: "{{cluster}}-01"
disks:
  - "{{cluster}}-01"
license_codes:
  [[PRIVATE]]
aggrs:
  - { name: aggr1, node: cluster1-01, disk_count: 13, max_raid: 26 }
inters:
  - { name: intercluster_1, address: 192.168.0.121, netmask: 255.255.255.0, node: cluster1-01, port: e0M, ipspace: Default }
vservers:
  - { name: svm1_cluster1, aggr: aggr1, protocol: 'cifs,nfs' }
lifs:
  - { name: svm1_data1, vserver: svm1_cluster1, node: cluster1-01, port: e0a, protocol: 'cifs,nfs', address: 192.168.0.130, netmask: 255.255.255.0 }
gateway:
  - { vserver: svm1_cluster1, destination: 0.0.0.0/0, gateway: 192.168.0.1 }
vserver_dns:
  - { vserver: svm1_cluster1, dns_domains: demo.netapp.com, dns_nameservers: 192.168.0.253 }
cifs:
  - { vserver: svm1_cluster1, cifs_server_name: cifs1, domain: demo.netapp.com, force: true }
```
./roles/na_ontap_cluster_setup/tasks/main.yml
```yaml
# Goal:
#   Write a playbook to complete the below list of tasks from a single node's FQDN.
#     Get nodes on switch with common cluster name (cluster[0-9]-*)
#       This will be able to handle vSims and HW based on the LOD standards
#     Create cluster from these nodes

---
- name: "Cluster - {{ cluster }}"
  netapp.ontap.na_ontap_cluster:
    state: present
    cluster_name: "{{ cluster }}"
    time_out: 30
    hostname: "{{ cluster }}-01"
    username: "{{ netapp_username }}"
    password: "{{ netapp_password }}"
    https: true
    validate_certs: "{{ validate_certs }}"

- name: Cluster Mgmt LIF
  na_ontap_interface:
    state: present
    interface_name: "{{ cluster_mgmt.interface_name }}"
    home_port: "{{ cluster_mgmt.home_port }}"
    home_node: "{{ cluster_mgmt.home_node }}"
    ipspace: "{{ cluster_mgmt.ipspace }}"
    address: "{{ cluster_mgmt.address }}"
    netmask: "{{ subnet }}"
    hostname: "{{ node1_ip }}"
    username: "{{ netapp_username }}"
    password: "{{ netapp_password }}"
    https: true
    validate_certs: "{{ validate_certs }}"

- name: Create User
  na_ontap_user:
    state: present
    name: admin
    applications: ssh,console,http,ontapi,service-processor
    authentication_method: password
    role_name: admin
    vserver: "{{ cluster }}"
    hostname: "{{ netapp_hostname }}"
    username: "{{ netapp_username }}"
    password: "{{ netapp_password }}"
    https: true
    validate_certs: "{{ validate_certs }}"

```

### Expected Results

I expected the task to checked and no change made because it (the interface) had been created on a prior run.

### Actual Results

```console
user@jumphost:~/ontap-setup-automation$ ansible-playbook 1-single-node_setup.yml -e "@cluster1.yml"
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ****************************************************************************************************************************************************************************************************************************************************************************************************

TASK [na_ontap_cluster_setup : Cluster - cluster1] ******************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [na_ontap_cluster_setup : Cluster Mgmt LIF] ********************************************************************************************************************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NoneType: None
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error creating interface cluster1_mgmt: calling: network/ip/interfaces: got {'message': 'Duplicate IP address 192.168.0.111', 'code': '1376963'}."}

PLAY RECAP **********************************************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

user@jumphost:~/ontap-setup-automation$
```

**Comment by lonico** [comment]
We fixed a similar issue in 21.21.0

I see two versions of the ONTAP collection:
```
netapp.ontap      21.20.0
netapp.ontap      21.22.0
```
Not sure which one has precedence.

If you run with `-vvv`, we should be able to spot the version.

```
ansible-doc netapp.ontap.na_ontap_cluster
```
should also show the version  in the first output line.

For the lab, it may be faster to just install ansible-core and control the additional collections you want to install.   Ansible comes with a lot of preinstalled collections.

**Comment by JamesPGriffith** [comment]
I will take your advice concerning version conflicts. I just had a colleague stand over my virtual shoulder and point out that I have the incorrect IP defined for the cluster_mgmt LIF which is in fact a duplicate of the node_mgmt LIF. This is also why I'm trying to rely on I'll leave this to someone else to determine whether or not the observed handling of the situation/error is correct.

If the observed handling of the situation/error is correct, then this can be closed...
